### PR TITLE
Deprecate Python 2.6 and 3.3 before removal

### PR DESCRIPTION
**Pull request by hugovk**
It's a good idea to deprecate before removal (https://github.com/jmespath/jmespath.py/pull/166), to give downstream dependencies a heads up.

How would you like to deprecate?

Here's some ideas. Let me know if you want any of them removing or changing.  I assumed JMESPath 1.0 would be the version they'll be dropped in, but that of course can be changed.

Warn when importing
* Pro: deprecation is visible
* Con: maybe too visible

Warn when installing
* Pro: gets the message across, not too often
* Con: maybe you don't want any warning

Document in CHANGELOG
* The minimum needed

Also added `python_requires` for the current versions (2.6+, 3.3+). This would be useful whatever is decided.

### CODE REVIEW

**Comment by jeking3** [comment]
Given folks can lock to a previous release, is deprecation necessary?

**Comment by hugovk** [comment]
See https://github.com/jmespath/jmespath.py/pull/166#issuecomment-459443888:

> I would love to drop support for python2.6, but this would affect many downstream dependencies that use jmespath. If we're going to do this, we'd need to have a deprecation period first to give people a heads up.

**Comment by jeking3** [comment]
I'm pretty sure that it would not, since folks downstream running well-designed CI jobs will have already locked their builds to a specific major or major.minor which would keep them compatible with python 2.6 until they can move forward.

File: setup.py [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
 
 from setuptools import setup, find_packages
 
 
+if sys.version_info[:2] <= (2, 6) or ((3, 0) <= sys.version_info[:2] <= (3, 3)):
+    python_ver = '.'.join(str(x) for x in sys.version_info[:3])
+
+    warnings.warn('You are using Python {0}, which will no longer be supported in '
+                  'JMESPath 1.0'.format(python_ver),
+                  DeprecationWarning)
```
Comment (jamesls) [inline setup.py]: I would prefer we start with a `PendingDeprecationWarning`.  We can then document the deprecation process in docs before starting to deprecate.

Comment (hugovk) [reply setup.py]: Updated!

**Comment by hugovk** [comment]
Updated to resolve conflicts!

---

I'd suggest moving forward with this process, as you can see Python 2.6 and 3.3 are not available on Travis CI:

image

```console
$ curl -sSf -o python-2.6.tar.bz2 ${archive_url}
curl: (22) The requested URL returned error: 404 Not Found
Unable to download 2.6 archive. The archive may not exist. Please consider a different version.
```

```console
$ curl -sSf -o python-3.3.tar.bz2 ${archive_url}
curl: (22) The requested URL returned error: 404 Not Found
Unable to download 3.3 archive. The archive may not exist. Please consider a different version.
```

https://travis-ci.org/jmespath/jmespath.py/builds/569774974

---

Also, Python 3.4 is also now EOL  and no longer receiving security updates (or any updates) from the core Python team.

Version | Release date | Supported until
--- | ---------- | ----------
2.7 | 2010-07-03 | 2020-01-01
3.4 | 2014-03-16 | 2019-03-16

I suggest deprecating/removing that as well.

**Comment by and-semakin** [comment]
Wow, Python 2.6 has reached its EOL in October, 2013. More then six years ago! I hope that all folks who uses it are already well prepared. I think that it (at least 2.6) can be dropped even without deprecation warning.

**Comment by jamesls** [comment]
I know it's been a while, but I want to follow up on this.  Here's my plan.

I think it's fine to move fairly quickly on deprecating 2.6/3.3, especially given some of the largest downstream dependencies have dropped 2.6/3.3 recently.  Initially I was surprised at the large number of 2.6 downloads, but in looking into it more, it seems to closely match the 2.6 numbers of the aws-cli/botocore which has already dropped 26/33 support.

```
| category | percent | downloads  |
|----------|--------:|-----------:|
| 2.7      |  51.87% | 22,981,642 |
| 3.7      |  16.73% |  7,411,660 |
| 3.6      |  13.75% |  6,092,117 |
| 2.6      |   6.98% |  3,090,928 |
| 3.5      |   5.05% |  2,237,316 |
| 3.8      |   3.34% |  1,481,817 |
| 3.4      |   1.77% |    785,991 |
| null     |   0.50% |    222,783 |
| 3.9      |   0.00% |        482 |
| 3.3      |   0.00% |        222 |
| 3.2      |   0.00% |          4 |
| Total    |         | 44,304,962 |

Date range: 2020-01-01 - 2020-01-31
```

So here's my plan:

* I'm going to get a 0.9.5 release out that adds support for py38
* I've created (#194) that fixes travis for 2.6.  I'll merge that and then rebase this PR with those changes.
* I'll cut 0.10.0 that officially deprecates 2.6/3.3.
* The next version, 0.11.0, will remove support for 2.6/3.3.  I'll give this about a month, so I'll plan on a release March 30th.

Let me know if there's any questions/concerns.

**Pull request closed without merging**